# Karan_Bhanushali_ABDAS_13_Statistics_for_Decision Making

## Importing the Data

In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

In [2]:
# Load the CSV file
csv_file_path = (r"C:\Users\ADMIN\Python\05. Assignment\Property Assignment\property.csv")
df_prop = pd.read_csv(csv_file_path)

In [3]:
# Checking column information
# Checking the missing values
missing_values = df_prop.isnull().sum()

# Get the data types
data_types = df_prop.dtypes

# Count duplicates
duplicate_count = df_prop.duplicated().sum()

# Create a summary DataFrame
summary = pd.DataFrame({
    'Missing Values': missing_values,
    'Data Types': data_types,
    'Duplicate Count': [duplicate_count] * len(missing_values)
})

summary

Missing Values Data Types  Duplicate Count
Suburb                      0     object                0
Address                     0     object                0
Rooms                       0      int64                0
Type                        0     object                0
Price                       0    float64                0
Method                      0     object                0
SellerG                     0     object                0
Date                        0     object                0
Distance                    0    float64                0
Postcode                    0    float64                0
Bedroom2                    0    float64                0
Bathroom                    0    float64                0
Car                        62    float64                0
Landsize                    0    float64                0
BuildingArea             6450    float64                0
YearBuilt                5375    float64                0
CouncilArea              1369     object                0
Lattitude                   0    float64                0
Longtitude                  0    float64                0
Regionname                  0     object                0
Propertycount               0    float64                0

In [4]:
# Check skewness of the 'Car' column
skewness_car = df_prop['Car'].skew()
print(f"Skewness of 'Car' column: {skewness_car}")

# Check skewness of the 'BuildingArea' column
skewness_building_area = df_prop['BuildingArea'].skew()
print(f"Skewness of 'BuildingArea' column: {skewness_building_area}")

# Check skewness of the 'YearBuilt' column
skewness_year_built = df_prop['YearBuilt'].skew()
print(f"Skewness of 'YearBuilt' column: {skewness_year_built}")

Skewness of 'Car' column: 1.3696759255976967
Skewness of 'BuildingArea' column: 77.69154091544215
Skewness of 'YearBuilt' column: -1.5412787597590965


In [5]:
# Fill missing values for numerical columns using the median
df_prop['Car'].fillna(df_prop['Car'].median(), inplace=True)
df_prop['BuildingArea'].fillna(df_prop['BuildingArea'].median(), inplace=True)
df_prop['YearBuilt'].fillna(df_prop['YearBuilt'].median(), inplace=True)

# Replace missing values in 'CouncilArea' column with 'Not Applicable'
df_prop['CouncilArea'].fillna('Not Applicable', inplace=True)

In [6]:
# Checking column information
# Checking the missing values
missing_values = df_prop.isnull().sum()

# Get the data types
data_types = df_prop.dtypes

# Count duplicates
duplicate_count = df_prop.duplicated().sum()

# Create a summary DataFrame
summary = pd.DataFrame({
    'Missing Values': missing_values,
    'Data Types': data_types,
    'Duplicate Count': [duplicate_count] * len(missing_values)
})

summary

Missing Values Data Types  Duplicate Count
Suburb                      0     object                0
Address                     0     object                0
Rooms                       0      int64                0
Type                        0     object                0
Price                       0    float64                0
Method                      0     object                0
SellerG                     0     object                0
Date                        0     object                0
Distance                    0    float64                0
Postcode                    0    float64                0
Bedroom2                    0    float64                0
Bathroom                    0    float64                0
Car                         0    float64                0
Landsize                    0    float64                0
BuildingArea                0    float64                0
YearBuilt                   0    float64                0
CouncilArea                 0     object                0
Lattitude                   0    float64                0
Longtitude                  0    float64                0
Regionname                  0     object                0
Propertycount               0    float64                0

# Q1

- For the suburb Altona, it is postulated that a typical property sells for 800000.
- Use the data at hand to test this assumption. Is the typical property price really $800,000 or has it increased?
- Use a significance level of 5%. 

In [7]:
from scipy.stats import ttest_1samp

# Filter data for the suburb "Altona" and extract property prices
altona_prices = df_prop[df_prop['Suburb'] == 'Altona']['Price'].dropna()

# Test value
test_value = 800000

# Perform a one-sample t-test
t_stat, p_value = ttest_1samp(altona_prices, test_value)

# Since it's a one-tailed test, divide the p-value by 2
p_value_one_tailed = p_value / 2

# Calculate mean price for reference
mean_price_altona = altona_prices.mean()

# Display results
print(f"Mean Property Price in Altona: ${mean_price_altona:,.2f}")
print(f"Test Statistic (t): {t_stat:.3f}")
print(f"One-Tailed p-Value: {p_value_one_tailed:.3f}")

# Conclusion
if p_value_one_tailed < 0.05:
    print("Conclusion: There is significant evidence that the typical property price in Altona has increased beyond $800,000.")
else:
    print("Conclusion: There is insufficient evidence to conclude that the typical property price in Altona has increased beyond $800,000.")

Mean Property Price in Altona: $834,830.41
Test Statistic (t): 1.028
One-Tailed p-Value: 0.154
Conclusion: There is insufficient evidence to conclude that the typical property price in Altona has increased beyond $800,000.


Based on the analysis, the mean property price in Altona is 834,830.41, with a test statistic of 1.028 and a one-tailed p-value of 0.154.
Given that the p-value exceeds the 5% significance level (0.05), we fail to reject the null hypothesis.
This indicates that there is insufficient statistical evidence to conclude that the typical property price in Altona has increased beyond 800,000. Therefore, the assumption that the typical property price is 800,000 remains plausible.

# Q2

- For the year 2016, is there any difference in prices of properties sold in the summer months vs winter months?
- Consider months from October till March as winter months and the rest as summer months. Use a significance level of 5%. 

In [8]:
from scipy.stats import ttest_ind

# Converting the 'Date' column to datetime format for proper filtering
df_prop['Date'] = pd.to_datetime(df_prop['Date'], format='%d/%m/%Y', errors='coerce')

# Filtering data for the year 2016
data_2016 = df_prop[df_prop['Date'].dt.year == 2016]

# Categorizing into summer and winter months
summer_months = [4, 5, 6, 7, 8, 9]  # April to September
winter_months = [10, 11, 12, 1, 2, 3]  # October to March

# Extracting prices for summer and winter months
summer_prices = data_2016[data_2016['Date'].dt.month.isin(summer_months)]['Price'].dropna()
winter_prices = data_2016[data_2016['Date'].dt.month.isin(winter_months)]['Price'].dropna()

# Perform a two-sample t-test
t_stat, p_value = ttest_ind(summer_prices, winter_prices)

# Display results
print(f"Mean Price (Summer): ${summer_prices.mean():,.2f}")
print(f"Mean Price (Winter): ${winter_prices.mean():,.2f}")
print(f"Test Statistic (t): {t_stat:.3f}")
print(f"p-Value: {p_value:.3f}")

# Conclusion
if p_value < 0.05:
    print("Conclusion: There is a significant difference in property prices between summer and winter months in 2016.")
else:
    print("Conclusion: There is no significant difference in property prices between summer and winter months in 2016.")

Mean Price (Summer): $1,048,054.73
Mean Price (Winter): $1,116,647.59
Test Statistic (t): -4.043
p-Value: 0.000
Conclusion: There is a significant difference in property prices between summer and winter months in 2016.


## Based on the independent two-sample t-test, the results indicate the following:

- The mean price of properties sold during summer months (April–September) is compared to the mean price of properties sold during winter months (October–March) in 2016.
- If the p-value from the test is less than 0.05, it suggests there is a statistically significant difference in property prices between the two periods.
- If the p-value is greater than 0.05, it means there is no statistically significant difference in property prices between summer and winter months.

## The exact conclusion:

- If p<0.05: There is enough evidence to conclude that property prices differ significantly between summer and winter months in 2016.
- If p≥0.05: There is insufficient evidence to conclude that property prices differ significantly between summer and winter months in 2016.

## Conclusion:
- Given that the p-value is less than the 5% significance level (0.05), we reject the null hypothesis.
- There is a statistically significant difference in property prices between the summer and winter months.
- Properties sold in winter months tend to have higher prices compared to those sold in summer months.

# Q3

- For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have car parking?
- Use the column car in the dataset. Round off your answer to 3 decimal places. 

In [9]:
from scipy.stats import binom

# Filtering data for the suburb "Abbotsford" and calculate probability of no car parking
abbotsford_data = df_prop[df_prop['Suburb'] == 'Abbotsford']
no_car_parking_prob = (abbotsford_data['Car'] == 0).mean()

# Given values for the binomial distribution
n = 10  # Total properties sold
k = 3   # Number of properties with no car parking

# Calculate binomial probability
probability = binom.pmf(k, n, no_car_parking_prob)

# Results
print(f"Probability of no car parking (p): {no_car_parking_prob:.3f}")
print(f"Probability that 3 out of 10 properties sold will not have car parking: {probability:.3f}")

Probability of no car parking (p): 0.268
Probability that 3 out of 10 properties sold will not have car parking: 0.260


## Conclusion:
- The probability that 3 out of 10 properties sold in Abbotsford will not have car parking is 0.260.
- This means there is a 26.0% chance that exactly 3 of the 10 properties sold will lack car parking.

# Q4

- In the suburb of Abbotsford, what are the chances of finding a property with 3 rooms?
- Round your answer to 3 decimal places. 

In [10]:
# Filter data for the suburb "Abbotsford"
abbotsford_data = df_prop[df_prop['Suburb'] == 'Abbotsford']

# Calculate the total number of properties and those with 3 rooms
total_properties = len(abbotsford_data)
properties_with_3_rooms = len(abbotsford_data[abbotsford_data['Rooms'] == 3])

# Calculate probability
probability = properties_with_3_rooms / total_properties

# Result
print(f"Probability of finding a property with 3 rooms in Abbotsford: {probability:.3f}")

Probability of finding a property with 3 rooms in Abbotsford: 0.357


## Conclusion:
- The probability of finding a property with 3 rooms in Abbotsford is 0.357.
- This indicates that there is a 35.7% chance of a randomly selected property in Abbotsford having 3 rooms.

# Q5 

- In the suburb of Abbotsford, what are the chances of finding a property with 2 bathrooms?
- Round your answer to 3 decimal places. 

In [11]:
# Filter data for the suburb "Abbotsford"
abbotsford_data = df_prop[df_prop['Suburb'] == 'Abbotsford']

# Calculate the total number of properties and those with 2 bathrooms
total_properties = len(abbotsford_data)
properties_with_2_bathrooms = len(abbotsford_data[abbotsford_data['Bathroom'] == 2])

# Calculate probability
probability = properties_with_2_bathrooms / total_properties

# Display results
print(f"Probability of finding a property with 2 bathrooms in Abbotsford: {probability:.3f}")

Probability of finding a property with 2 bathrooms in Abbotsford: 0.339


## Conclusion:
- The probability of finding a property with 2 bathrooms in Abbotsford is 0.339.\
- This means there is a 33.9% chance that a randomly selected property in Abbotsford will have 2 bathrooms.